src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+import { posix } from 'path';
+import { types } from 'mime-types';
+import { RuntimeConfig } from '../init/RuntimeConfig';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { APPLICATION_OCTET_STREAM, TEXT_TURTLE } from '../util/ContentTypes';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { trimTrailingSlashes } from '../util/Util';
+import { FileIdentifierMapper } from './FileIdentifierMapper';
+
+const { join: joinPath, normalize: normalizePath } = posix;
+
+export interface PathAbstraction {
+
+  /**
+   * The parent directory path of the file.
+   */
+  path: string;
+
+  /**
+   * The name/slug of the file.
+   */
+  slug?: string;
+}
+
+export class ExtensionBasedMapper implements FileIdentifierMapper {
+  private readonly runtimeConfig: RuntimeConfig;
+  private readonly types: Record<string, any>;
+
+  public constructor(runtimeConfig: RuntimeConfig, overrideTypes = { acl: TEXT_TURTLE, metadata: TEXT_TURTLE }) {
+    this.runtimeConfig = runtimeConfig;
+    this.types = { ...types, ...overrideTypes };
```

Comment:
That's a gigantic join every time (but I suppose it only really happens once in practice).

Reply:
Maybe this could be avoided by using the content type function from mime-types, and if no matching content type is found, it should check in the overrideTypes (so it would make getContentTypeFromExtension a bit longer then).

Reply:
Indeed, but I think it's fine now. Will be faster/easier this way, and likely not a bottleneck.